lib/src/style/layer/background_layer.dart

```diff
@@ -80,7 +80,9 @@ class BackgroundLayer extends Layer {
               .toLowerCase()
               .contains(map["layout"]["visibility"])),
       backgroundColor: (map["paint"]["background-color"] as List?)?.toRGBAInt(),
-      backgroundOpacity: map["paint"]["background-opacity"]?.toDouble(),
+      backgroundOpacity: map["paint"]["background-opacity"] is double
```

Comment:
just a question. is it possible to write some extensions like in kotlin?

```
getDoubleOrNull(map["paint"]["background-opacity"]){
  if this is double return this else null
}
```

Reply:
Yeah we can add an extension but I wouldn't optimize it since it's just a generated code